server/catgenome/src/main/java/com/epam/catgenome/manager/reference/ReferenceManager.java

```diff
@@ -544,6 +559,10 @@ private String parse(final String path, final String name) {
     private long registerReference(Long referenceId, Reference reference, boolean createGC)
             throws IOException {
         String path = reference.getPath();
+        if (GenbankUtils.isGenbank(path)) {
+            path = genbankToFasta(reference);
```

Comment:
There is a problem with such approach, because when you set here new path, you already register biologicalDataItem with another path in line `188`, so I believe we should move this peace of code before BDI registration